src/System.Buffers/src/System/Buffers/NativeBufferPool.cs

```diff
@@ -21,6 +40,13 @@ public NativeBufferPool(int maxElementsInBuffer)
             _buckets = new NativeBufferBucket<T>[maxBuckets + 1];
         }
 
+        public void Dispose()
+        {
+            foreach (NativeBufferBucket<T> bucket in _buckets)
```

Comment:
The only reason I can think of where the Interlock would be needed is during a double-dispose problem, in which case the caller has a logic error in their calling pattern anyway. The Dispose function should be called, at most, twice; once via the explicit Dispose call from the caller (by a ```using``` block or an explicit call) and once via the Finalizer, which will pass in False to the ```Dispose``` function, meaning nothing will happen. Having an interlocked operation and setting to null shouldn't be necessary